server/db/dbManager.ts

```diff
@@ -24,26 +24,34 @@ export class PostManager {
 		return this.model.findById(postId).exec();
 	};
 
-	async getAllPosts(): Promise<models.IPost[]> {
-		return this.model.find({}).exec();
+	async getAllPosts(parent?: mongoose.Types.ObjectId): Promise<models.IPost[]> {
+		let posts;
+		if (parent) {
+			posts = this.model.find({ parent });
+		} else {
+			posts = this.model.find({ parent: undefined });
+		}
```

Comment:
I'm a bit confused. `parent` can be `mongoose.Types.ObjectId` or `undefined` only. So this what you can pass accordingly. I'd remove this `if` clause at all and simplify the next lines.

Reply:
I see what you mean, this is actually a nonissue in the next revision of getAllPosts